helm/data_template.go

```diff
@@ -0,0 +1,594 @@
+	client.Description = d.Get("description").(string)
+	client.CreateNamespace = d.Get("create_namespace").(bool)
+
+	// The following source has been adapted from the source of the helm template command
+	// https://github.com/helm/helm/blob/v3.5.3/cmd/helm/template.go#L67
+	client.DryRun = true
+	// Do not set fixed release name like in helm template
+	//client.ReleaseName = "RELEASE-NAME"
+	client.Replace = true // Skip the name check
+	client.ClientOnly = !d.Get("validate").(bool)
+	client.APIVersions = chartutil.VersionSet(apiVersions)
+	client.IncludeCRDs = d.Get("include_crds").(bool)
+
+	skipTests := d.Get("skip_tests").(bool)
+
+	debug("%s Rendering Chart", logID)
+
+	rel, err := client.Run(c, values)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var manifests bytes.Buffer
+
+	fmt.Fprintln(&manifests, strings.TrimSpace(rel.Manifest))
+
+	if !client.DisableHooks {
+		for _, m := range rel.Hooks {
+			if skipTests && isTestHook(m) {
+				continue
+			}
+
+			fmt.Fprintf(&manifests, "---\n# Source: %s\n%s\n", m.Path, m.Manifest)
+		}
+	}
+
+	// Difference to the implementation of helm template in newTemplateCmd:
+	// Independent of templates, names of the charts templates are always resolved from the manifests
+	// to be able to populate the keys in the manifests computed attribute.
+	var manifestsToRender []string
+
+	splitManifests := releaseutil.SplitManifests(manifests.String())
+	manifestsKeys := make([]string, 0, len(splitManifests))
+	for k := range splitManifests {
+		manifestsKeys = append(manifestsKeys, k)
+	}
+	sort.Sort(releaseutil.BySplitManifestsOrder(manifestsKeys))
+
+	// Mapping of manifest key to manifest template name
+	manifestNamesByKey := make(map[string]string, len(manifestsKeys))
+
+	manifestNameRegex := regexp.MustCompile("# Source: [^/]+/(.+)")
+
+	for _, manifestKey := range manifestsKeys {
+		manifest := splitManifests[manifestKey]
+		submatch := manifestNameRegex.FindStringSubmatch(manifest)
+		if len(submatch) == 0 {
+			continue
+		}
+		manifestName := submatch[1]
+		manifestNamesByKey[manifestKey] = manifestName
+	}
+
+	// if we have a list of files to render, then check that each of the
+	// provided files exists in the chart.
+	if len(templates) > 0 {
+		for _, f := range templates {
+			missing := true
+			// Use linux-style filepath separators to unify user's input path
+			f = filepath.ToSlash(f)
+			for manifestKey, manifestName := range manifestNamesByKey {
+				// manifest.Name is rendered using linux-style filepath separators on Windows as
+				// well as macOS/linux.
+				manifestPathSplit := strings.Split(manifestName, "/")
+				// manifest.Path is connected using linux-style filepath separators on Windows as
+				// well as macOS/linux
+				manifestPath := strings.Join(manifestPathSplit, "/")
+
+				// if the filepath provided matches a manifest path in the
+				// chart, render that manifest
+				if matched, _ := filepath.Match(f, manifestPath); !matched {
```

Comment:
Same as above and I confess: "mostly copied" ;)

When rebasing this PR to the latest helm revision, it was easier to reflect the upstream changes, because I did not refactor the original implementation.